aide_design/unit_process_design/prefab/lfom_prefab_functional.py

```diff
@@ -37,157 +34,163 @@
 def width_stout(HL_LFOM,z):
-    return 2/((2*u.g_0*z)**(1/2)*math.pi*HL_LFOM)
-
-def n_lfom_rows(Q,HL_LFOM):
-    N_estimated = (HL_LFOM*math.pi/(2*width_stout(HL_LFOM,HL_LFOM)*Q)).to(u.dimensionless)
-    return min(10,max(4,math.trunc(N_estimated.magnitude)))
-
-FLOW = 5*u.L/u.s
-HL_LFOM = 20*u.cm
+    return (2/((2*pc.gravity*z)**(1/2)*ratio_VC_orifice*np.pi*HL_LFOM)).magnitude
+
+
+@u.wraps(None, [u.m**3/u.s,u.m], False)
+def n_lfom_rows(FLOW,HL_LFOM):
+    """This equation states that the open area corresponding to one row can be
+    set equal to two orifices of diameter=row height. If there are more than 
+    two orifices per row at the top of the LFOM then there are more orifices 
+    than are convenient to drill and more than necessary for good accuracy. 
+    Thus this relationship can be used to increase the spacing between the 
+    rows and thus increase the diameter of the orifices. This spacing function 
+    also sets the lower depth on the high flow rate LFOM with no accurate 
+    flows below a depth equal to the first row height.
+    
+    But it might be better to always set then number of rows to 10.
+    The challenge is to figure out a reasonable system of constraints that
+    reliably returns a valid solution.
+    """
+    N_estimated = (HL_LFOM*np.pi/(2*width_stout(HL_LFOM,HL_LFOM)*FLOW))
+    #variablerow=min(10,max(4,math.trunc(N_estimated.magnitude)))
+    return 10
 
 
 def dist_center_lfom_rows(FLOW,HL_LFOM):
     return HL_LFOM/n_lfom_rows(FLOW,HL_LFOM)
 
 
-
-# average vertical velocity of the water inside the LFOM pipe 
-# at the very bottom of the bottom row of orifices
-# The speed of falling water is 0.841 m/s for all linear flow orifice meters of height 20cm, independent of total plant flow rate.
-def vol_lfom_pipe_critical(HL_LFOM):
+def vel_lfom_pipe_critical(HL_LFOM):
+    """The average vertical velocity of the water inside the LFOM pipe 
+    at the very bottom of the bottom row of orifices
+    The speed of falling water is 0.841 m/s for all linear flow orifice meters
+    of height 20 cm, independent of total plant flow rate."""
     return (4/(3*math.pi)*(2*u.g_0*HL_LFOM)**(1/2)).to(u.m/u.s)
 
 def area_lfom_pipe_min(FLOW,HL_LFOM,Pi_LFOM_safety):
-    return (Pi_LFOM_safety*FLOW/vol_lfom_pipe_critical(HL_LFOM)).to(u.m**2)
+    return (Pi_LFOM_safety*FLOW/vel_lfom_pipe_critical(HL_LFOM)).to(u.m**2)
 
 def nom_diam_lfom_pipe(FLOW,HL_LFOM,Pi_LFOM_safety,SDR_LFOM):
     ID=pc.diam_circle(area_lfom_pipe_min(FLOW,HL_LFOM,Pi_LFOM_safety))
     return pipe.ND_SDR_available(ID,SDR_LFOM)
 
-# another possibility is to use integration to solve this problem.
-# Here we use the width of the stout weir in the center of the top row
-# to estimate the area of the top orifice
 def area_lfom_orifices_max(FLOW,HL_LFOM):
-    return FLOW*width_stout(HL_LFOM,HL_LFOM-0.5*dist_center_lfom_rows(FLOW,HL_LFOM))*dist_center_lfom_rows(FLOW,HL_LFOM)
+    """Estimate the orifice area corresponding to the top row of orifices.
```

Comment:
Is anything guaranteed about this estimate? Estimate of is a very unclear descriptor. What can we be guaranteed we'll get out of this function? Also, return needs to return a single variable and this is hard to understand. Might be helpful to break it into smaller pieces: 
```
((FLOW*width_stout(HL_LFOM,HL_LFOM-0.5*dist_center_lfom_rows(FLOW,HL_LFOM))*dist_center_lfom_rows(FLOW,HL_LFOM))).to(u.m**2)
```